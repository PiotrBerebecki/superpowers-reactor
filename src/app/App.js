import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import './App.css';
import Header from './Header';
import Puzzles from './../views/Puzzles';

class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Header />
          <Route exact path="/" render={() => <Puzzles category="latest" />} />
          <Route
            path="/:category"
            render={({ match }) => <Puzzles category={match.params.category} />}
          />
        </div>
      </Router>
    );
  }
}

export default App;
