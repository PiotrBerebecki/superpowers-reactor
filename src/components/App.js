import React, { Component } from 'react';

import './App.css';

class App extends Component {
  render() {
    return <div>Superpowers Reactor</div>;
  }
}

export default App;
