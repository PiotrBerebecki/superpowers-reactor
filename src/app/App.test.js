import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import App from './App.js';

describe('App', () => {
  it('renders component correctly', () => {
    const component = renderer.create(<App />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes relevant prop to home view', () => {
    const component = shallow(<App />);
    const homeRoute = component
      .find('Route')
      .findWhere(route => route.props().path === '/');

    const homeComponent = homeRoute.props().render();
    expect(homeComponent.props.category).toEqual('latest');
  });

  it('passes relevant prop to other views', () => {
    const component = shallow(<App />);
    const otherRoute = component
      .find('Route')
      .findWhere(route => route.props().path === '/:category');

    const otherComponent = otherRoute
      .props()
      .render({ match: { params: { category: 'trending' } } });
    expect(otherComponent.props.category).toEqual('trending');
  });
});
