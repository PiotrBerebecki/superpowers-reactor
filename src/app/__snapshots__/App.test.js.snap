// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  <div>
    <a
      className="sc-bdVaJa gRwgNw"
      href="/latest"
      onClick={[Function]}
      style={undefined}
    >
      latest
    </a>
    <a
      className="sc-bdVaJa gRwgNw"
      href="/trending"
      onClick={[Function]}
      style={undefined}
    >
      trending
    </a>
    <a
      className="sc-bdVaJa gRwgNw"
      href="/top"
      onClick={[Function]}
      style={undefined}
    >
      top
    </a>
  </div>
</div>
`;
